api/filter/http_fault.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto3";
+
+package envoy.api.v2.filter;
+
+import "api/rds.proto";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/wrappers.proto";
+
+// Delay specification is used to inject latency into the rpc operations.
+message Delay {
+  // Delay type to use (fixed|exponential|..). Currently, only fixed delay (step function) is supported.
+  string type = 1;
+
+  // An integer between 0-100 indicating the percentage of operations
+  // on which the delay will be injected.
+  google.protobuf.UInt32Value percent = 2;
+
+  oneof http_delay_type {
+    // Add a fixed delay before forwarding the operation upstream. Format: 1h/1m/1s/1ms. MUST be >=1ms.
+    google.protobuf.Duration fixed_delay = 2;
+    // (-- Add a delay (based on an exponential function) before forwarding
+    // the operation. Mean delay needed to derive the exponential delay
+    // values. NOT IMPLEMENTED --)
+    google.protobuf.Duration exponential_delay = 3 ;
+  }
+  // (-- Specify delay duration as part of Http request. NOT IMPLEMENTED --)
+  string override_header_name = 4 ;
+}
+
+// HTTPFault can be used to specify one or more faults to inject
```

Comment:
Are you writing docs from scratch? In general, I've been borrowing heavily from the existing stuff at places like https://envoyproxy.github.io/envoy/configuration/http_filters/fault_filter.html#configuration.

Reply:
Copied in bits and pieces from Istio specs.

Reply:
Can you try and reconcile with the existing docs? I don't mind if there is additional material in the filter descriptions, but we should try not to be lossy.

Reply:
synced with docs..